src/modules/jcms_article/jcms_article.module

```diff
@@ -32,8 +32,10 @@ function jcms_article_node_presave(EntityInterface $entity) {
     return NULL;
   }
   $node_presave = \Drupal::service('jcms_article.hooks.node_presave');
-  $node_presave->addJsonFields($entity);
-  $node_presave->setPublishedStatus($entity);
-  $node_presave->setStatusDate($entity);
-  $node_presave->setSubjectTerms($entity);
+  // TODO: retrieve with getArticleById() and pass it in
+  $article = $node_presave->getArticleById($entity->label());
+  $node_presave->addJsonFields($entity, $article);
+  $node_presave->setPublishedStatus($entity, $article);
+  $node_presave->setStatusDate($entity, $article);
+  $node_presave->setSubjectTerms($entity, $article);
```

Comment:
Since we don't know for how many hours/days the NodePresave objects has been around, anything cached inside the object or static can in principle be stale. Moreover, storing responses bloats memory (I get daily alerts `Memory > 90% on 'journal-cms--prod--1` at the moment). An ideal way to avoid the multiple parameter passing would be to create a new NodePresave object for each call to this method.